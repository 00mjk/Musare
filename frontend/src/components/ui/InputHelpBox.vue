<template>
	<p class="help" :class="valid ? 'is-success' : 'is-danger'">
		{{ message }}
	</p>
</template>

<script>
export default {
	props: {
		message: {
			type: String,
			required: true
		},
		valid: {
			type: Boolean,
			required: true
		}
	},
	data() {
		return {};
	},
	methods: {}
};
</script>

<style lang="scss" scoped>
.help {
	margin-top: 0 !important;
	margin-bottom: 5px !important;
	font-size: 12px;
}
</style>
