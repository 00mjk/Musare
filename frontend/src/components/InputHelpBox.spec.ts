import { mount } from "@vue/test-utils";
import InputHelpBox from "@/components/InputHelpBox.vue";

test("input help box component props", async () => {
	expect(InputHelpBox).toBeTruthy();

	const wrapper = mount(InputHelpBox, {
		props: {
			message: "This input has not been entered and is valid.",
			valid: true,
			entered: false
		}
	});

	expect(wrapper.text()).toBe(
		"This input has not been entered and is valid."
	);
	expect(wrapper.classes()).toContain("is-grey");

	await wrapper.setProps({
		message: "This input has not been entered and is invalid.",
		valid: false,
		entered: false
	});
	expect(wrapper.text()).toBe(
		"This input has not been entered and is invalid."
	);
	expect(wrapper.classes()).toContain("is-grey");

	await wrapper.setProps({
		message: "This input has been entered and is valid.",
		valid: true,
		entered: true
	});
	expect(wrapper.text()).toBe("This input has been entered and is valid.");
	expect(wrapper.classes()).toContain("is-success");

	await wrapper.setProps({
		message: "This input has potentially been entered and is valid.",
		valid: true,
		entered: undefined
	});
	expect(wrapper.text()).toBe(
		"This input has potentially been entered and is valid."
	);
	expect(wrapper.classes()).toContain("is-success");

	await wrapper.setProps({
		message: "This input has been entered and is invalid.",
		valid: false,
		entered: true
	});
	expect(wrapper.text()).toBe("This input has been entered and is invalid.");
	expect(wrapper.classes()).toContain("is-danger");

	await wrapper.setProps({
		message: "This input has potentially been entered and is invalid.",
		valid: false,
		entered: undefined
	});
	expect(wrapper.text()).toBe(
		"This input has potentially been entered and is invalid."
	);
	expect(wrapper.classes()).toContain("is-danger");

	expect(wrapper.html()).toMatchSnapshot();
	expect(wrapper.classes()).toContain("help");
});
