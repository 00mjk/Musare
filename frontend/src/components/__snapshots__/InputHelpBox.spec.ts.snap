// Vitest Snapshot v1

exports[`input help box component props 1`] = `"<p class=\\"help is-danger\\" data-v-109db604=\\"\\">This input has potentially been entered and is invalid.</p>"`;
